### DISCUSSION TITLE
Bun support

### DISCUSSION THREAD

**Discussion by alyiev** [expo/expo#23931] [Proposals and Ideas]
Bun will get a stable version in September. Are there any plans to support it?

**Comment by TheOmran** [comment]
Bun stable version v1.0 is now out and production ready, would definitely be great to support it.

**Comment by tsapeta** [comment]
Our CLIs already do work just fine with Bun. We're looking into using it on EAS builders.

**Reply by marcosflp** [reply]
Hey @tsapeta, thanks for the feedback.

I tried using it with an existing Expo project, but I'm getting the error:
<img width="341" alt="Captura de Tela 2023-09-11 às 19 25 18" src="https://github.com/expo/expo/assets/6920815/ad717b62-a0c6-4849-be05-e55f1e0e99e8">

I follow these steps to install and run it:
```
$ rm -rf node_modules yarn.lock
$ bun install
$ bun start
```

Looks like it's missing `node`, but I thought Bun would replace `node` and would not be required to run the app. If I activate the nvm like `nvm use v16.17.0` it works fine.

Does anyone know what is happening?

**Reply by tsapeta** [reply]
- `expo start` uses the global CLI which is deprecated
- some processes that are spawned from our CLIs (like Metro bundler) are spawned using Node, because we are not yet sure if they work correctly with Bun, but in general our own code runs on Bun

**Reply by imaf6971** [reply]
How can I get it to run fully on bun? Just for testing purposes

**Comment by iM-GeeKy** [comment]
This felt like the appropriate spot to ask this question, but is there an option to force `eas build` to use `yarn` if we are using `bun install --yarn`. According to the documentation if there is a `bun.lockb` `eas build` will automatically use `bun`. The reason I ask this and perhaps it's a problem on my end, is my builds do not work when `eas build` uses `bun` to install dependences, etc, but if I delete the `bun.lockb` and leave the `yarn.lock` my builds work as expected.

Here are my builds in case that may be helpful:

- Failing with bun (https://expo.dev/accounts/bluebirding/projects/bluebird-client/builds/bfe93a86-4873-4674-abc9-50df04434e27)
- Successful w/ yarn (https://expo.dev/accounts/bluebirding/projects/bluebird-client/builds/da53c04d-43bf-44f6-955f-788fec5f6eee)

**Reply by tsapeta** [reply]
cc @byCedric
